### PR TITLE
Refactor NavBar

### LINKED ISSUE #141: Refactor Navbar
## Description:
Currently, the NavBar is on every page, which causes a short flickering when navigating. We should move the NavBar out of the pages to prevent this behavior.

### PR DESCRIPTION
**Pull request by TomRomeo**
## Description
This PR refactors the navbar to the `_app.tsx` layout file so it doesn't get re-rendered on every page navigation.

## Resolved Issues
resolves #141

### CODE REVIEW

**Review by memeToasty — APPROVED** [review]

File: frontend/components/navigation/Navbar.tsx [hunk 1]
```diff
@@ -13,52 +14,56 @@ const inter = Inter({
...
  * UI component for diplaying the Navbar
  */
-export const Navbar = ({ login, className = "", onClick }: NavbarProps) => {
+export const Navbar = ({ className = "" }: NavbarProps) => {
+	const [loggedIn, setLoggedIn] = useState(false);
+	useEffect(() => {
+		setLoggedIn(hasCookie("access_token"));
+	}, []);
 	return (
 		<div
```
Comment (darmiel) [inline frontend/components/navigation/Navbar.tsx:32]: use `nav` instead of `div`

File: frontend/components/navigation/Navbar.tsx [hunk 2]
```diff
@@ -13,52 +14,56 @@ const inter = Inter({
...
-							if (response?.ok) {
-								location.reload();
-							}
-						}}
-					></LogOut>
-				</div>
+			{loggedIn ? (
+				<>
+					<Logo onClick={() => router.push("/")}></Logo>
+					<div className="flex flex-row">
```
Comment (darmiel) [inline frontend/components/navigation/Navbar.tsx:38]: why flex (-row) this?

File: frontend/pages/features.tsx [hunk 3]
```diff
@@ -3,23 +3,17 @@ import { useRouter } from "next/router";
...
 import { Award, BarChart2, Cloud, Code, Compass, Users } from "react-feather";
 
-import { Navbar } from "../components/navigation/Navbar";
-
 export default function Page() {
 	const router = useRouter();
 
 	return (
-		<div>
+		<div className="flex flex-1">
```
Comment (darmiel) [inline frontend/pages/features.tsx:10]: use `main` instead of `div`

**Review by darmiel — APPROVED** [review]

File: frontend/components/navigation/Navbar.tsx (hunk 1 above)
Comment (TomRomeo) [reply frontend/components/navigation/Navbar.tsx]: chaged in 6ed433b

File: frontend/components/navigation/Navbar.tsx (hunk 2 above)
Comment (TomRomeo) [reply frontend/components/navigation/Navbar.tsx]: removed in 6ed433b

File: frontend/pages/features.tsx (hunk 3 above)
Comment (TomRomeo) [reply frontend/pages/features.tsx]: changed in 6ed433b

**Review by Julian702 — CHANGES_REQUESTED** [review]
There are some small problems we have to fix:
- [x] The login button is not updated after login.
- [x] The Navbar is now displayed on the login page. That's why the logo is displayed twice and the login button in the Navbar is also not convenient on the login page. I would suggest removing the login button on the login page from the navbar and removing the logo from the login page. Moreover, the login page is not centered.

Furthermore, I don't really like the layout changes on the home page and we should try to restore the old layout after the refactoring.

File: frontend/components/navigation/Navbar.tsx [hunk 4]
```diff
@@ -13,53 +14,57 @@ const inter = Inter({
...
 
 /**
  * UI component for diplaying the Navbar
  */
-export const Navbar = ({ login, className = "", onClick }: NavbarProps) => {
+export const Navbar = ({ className = "" }: NavbarProps) => {
+	const [loggedIn, setLoggedIn] = useState(false);
+	useEffect(() => {
+		setLoggedIn(hasCookie("access_token"));
+	}, []);
```
[the remark this answers is not in the document]
Comment (TomRomeo) [reply frontend/components/navigation/Navbar.tsx]: refactored in d9cf355

File: frontend/components/navigation/Navbar.tsx [hunk 5]
```diff
@@ -13,53 +14,57 @@ const inter = Inter({
...
-							});
-							if (response?.ok) {
-								location.reload();
-							}
-						}}
-					></LogOut>
-				</div>
+			{loggedIn ? (
+				<>
+					<Logo onClick={() => router.push("/")}></Logo>
```
[the remark this answers is not in the document]
Comment (TomRomeo) [reply frontend/components/navigation/Navbar.tsx]: moved in d9cf355

File: frontend/components/navigation/Navbar.tsx [hunk 6]
```diff
@@ -13,53 +14,57 @@ const inter = Inter({
...
+					<div className="flex">
```
[the remark this answers is not in the document]
Comment (TomRomeo) [reply frontend/components/navigation/Navbar.tsx]: removed in d9cf355

File: frontend/components/navigation/Navbar.tsx [hunk 7]
```diff
@@ -13,53 +14,57 @@ const inter = Inter({
...
-				<Button
-					id="loginButton"
-					style="secondary"
-					className="invisible h-full justify-end sm:visible"
-					onClick={() => router.push("/")}
-				>
-					Login <ArrowRight className="ml-1 h-2/3"></ArrowRight>
-				</Button>
+				<>
+					<Logo onClick={() => router.push("/home")}></Logo>
```
[the remark this answers is not in the document]
Comment (TomRomeo) [reply frontend/components/navigation/Navbar.tsx]: moved in d9cf355

File: frontend/pages/features.tsx [hunk 8]
```diff
@@ -1,29 +1,24 @@
...
-				<Navbar
-					login={false}
-					onClick={() => router.push("/home")}
-				></Navbar>
+			<div className="min-w-screen flex flex-col bg-eggshell">
 				<div className="mx-auto flex flex-col justify-center p-5 text-base leading-7 md:w-2/3 md:p-10 md:text-center">
 					<a
 						className="text-lg font-semibold text-kiokuLightBlue hover:cursor-pointer"
-						onClick={() => router.push("/login")}
+						onClick={() => hasCookie("access-token") ? router.push("/") : router.push("/login")}
```
Comment (Julian702) [inline frontend/pages/features.tsx:21]: This does not work. The method `hasCookie` returns `false`. Even if I am logged in, I am redirected to the login by clicking on the link or the button on the home page.

File: frontend/pages/home.tsx [hunk 9]
```diff
@@ -34,9 +30,7 @@ export default function Page() {
 						<div className="flex flex-row space-x-3 md:space-x-5">
 							<Button
 								id="getstartedButton"
-								onClick={() => {
-									router.push("/");
-								}}
+								onClick={() => hasCookie("access-token") ? router.push("/") : router.push("/login")}
```
Comment (Julian702) [inline frontend/pages/home.tsx:33]: s.a.

**Review by Julian702 — CHANGES_REQUESTED** [review]
The getting startedg buttons are still not working as expected. Please review and test your changes afterwards. 
It is also possible to access the login page when logged in. In this case, the Logout button is displayed in the NavBar. We should change this behavior. I would suggest either to be redirected directly when you go to the login page logged in or to be logged out automatically.

File: frontend/pages/features.tsx (hunk 8 above)
Comment (TomRomeo) [reply frontend/pages/features.tsx]: This was due to the wrongly quoted cookie name. The actual name is `access_token`.
I fixed it in b4d3352

File: frontend/pages/home.tsx (hunk 9 above)
Comment (TomRomeo) [reply frontend/pages/home.tsx]: also fixed in b4d3352

**Comment by TomRomeo** [comment]
> [...] It is also possible to access the login page when logged in. In this case, the Logout button is displayed in the NavBar. We should change this behavior. I would suggest either to be redirected directly when you go to the login page logged in or to be logged out automatically.
I changed the behavior to redirect the user away from the login page when they are logged in.

**Review by Julian702 — APPROVED** [review]

File: frontend/components/navigation/Navbar.tsx [hunk 10]
```diff
@@ -1,6 +1,7 @@
+import { hasCookie } from "cookies-next";
 import { Inter } from "next/font/google";
```
Comment (4KevR) [inline frontend/components/navigation/Navbar.tsx:2]: Inter is not used anymore in this file. Remove import and usage.

File: frontend/components/navigation/Navbar.tsx [hunk 11]
```diff
@@ -1,6 +1,7 @@
...
-import router from "next/router";
-import React from "react";
+import router, { useRouter } from "next/router";
```
Comment (4KevR) [inline frontend/components/navigation/Navbar.tsx:3]: Unused import `router`

File: frontend/components/navigation/Navbar.tsx [hunk 12]
```diff
@@ -13,53 +14,57 @@ const inter = Inter({
...
+							router.replace("/home");
+						}
+					}}
+				></LogOut>
+			)}
+			{loggedIn == false && (
 				<Button
 					id="loginButton"
 					style="secondary"
 					className="invisible h-full justify-end sm:visible"
```
Comment (4KevR) [inline frontend/components/navigation/Navbar.tsx:62]:
```suggestion
					className="invisible justify-end sm:visible h-7 md:h-10"
```
The Login Button is causing a shift in the position of the Logo when changing from `home` to `login` page.

**Review by 4KevR — CHANGES_REQUESTED** [review]
Looks great, just some small nits.

File: frontend/components/navigation/Navbar.tsx (hunk 10 above)
Comment (TomRomeo) [reply frontend/components/navigation/Navbar.tsx]: removed in 8c37524

File: frontend/components/navigation/Navbar.tsx (hunk 11 above)
Comment (TomRomeo) [reply frontend/components/navigation/Navbar.tsx]: removed in 8c37524

File: frontend/components/navigation/Navbar.tsx (hunk 12 above)
Comment (TomRomeo) [reply frontend/components/navigation/Navbar.tsx]: changed in 8c37524

**Review by 4KevR — APPROVED** [review]

**Pull request merged**